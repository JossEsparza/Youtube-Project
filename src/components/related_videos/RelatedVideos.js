import React from 'react';
// import { useFetch } from '../../hooks/useFetch';
// import { useLocation } from 'react-router-dom';
import RelatedVideoCard from '../cards/RelatedVideoCard';
import './RelatedVideos.scss';

const RelatedVideos = () => {
  const data = require('../../mocks/youtube-related-videos-mock.json');
  const error = 0;
  const loading = 0;
  // const location = useLocation();
  // const key = 'apikey';

  // const { data, loading, error } = useFetch(
  //   'https://www.googleapis.com/youtube/v3/search?part=snippet&key=' +
  //     key +
  //     '&relatedToVideoId=' +
  //     location.pathname.substring(1) +
  //     '&type=video'
  // );

  // console.log(data.items);

  const filtered_video = data?.items?.filter((video) => {
    return video.snippet;
  });

  if (error) {
    return <p>{error.message}</p>;
  }

  return (
    <div>
      <div className="related_container">
        {loading ? (
          <p>Loading!</p>
        ) : (
          filtered_video.map((video) => (
            <RelatedVideoCard key={video.id.videoId} {...video} />
          ))
        )}
      </div>
    </div>
  );
};

export default RelatedVideos;
