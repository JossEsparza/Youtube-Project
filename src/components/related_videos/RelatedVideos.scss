.related_container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: left;
  width: 500px;
}
