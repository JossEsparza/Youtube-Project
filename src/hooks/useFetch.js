import { useState, useEffect } from 'react';

export const useFetch = (url) => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    let controller = new AbortController();

    const fetchData = async () => {
      setLoading(true);

      try {
        const response = await fetch(url, { signal: controller.signal });
        const result = await response.json();
        setData(result);
      } catch (error) {
        setError(error);
      }
      setLoading(false);
    };
    fetchData();

    return () => {
      controller.abort();
    };
  }, [url]);

  return {
    data,
    loading,
    error,
  };
};
