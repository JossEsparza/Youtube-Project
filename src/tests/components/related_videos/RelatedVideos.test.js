import { mount } from 'enzyme';
import { Route, MemoryRouter } from 'react-router-dom';
import RelatedVideos from '../../../components/related_videos/RelatedVideos';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');

describe('Tests the Related Videos', () => {
  test('should be displayed correctly when loading is true', () => {
    const mockLocation = {
      pathname: '/',
      search: '',
    };
    useFetch.mockReturnValue({
      data: {},
      loading: true,
      error: null,
    });
    const wrapper = mount(
      <MemoryRouter history={mockLocation}>
        <Route path="/:video_id">
          <RelatedVideos />
        </Route>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should be displayed correctly when loading is false', () => {
    const mockLocation = {
      pathname: '/',
      search: '',
    };
    useFetch.mockReturnValue({
      data: {
        items: [
          {
            snippet: {
              title: 'my first video',
              thumbnails: {
                default: {
                  url: 'https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg',
                },
              },
            },
          },
        ],
      },
      loading: false,
      error: null,
    });
    const wrapper = mount(
      <MemoryRouter history={mockLocation}>
        <Route path="/:video_id">
          <RelatedVideos />
        </Route>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
