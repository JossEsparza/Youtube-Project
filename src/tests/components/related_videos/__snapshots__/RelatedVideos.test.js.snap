// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Related Videos should be displayed correctly when loading is false 1`] = `ReactWrapper {}`;

exports[`Tests the Related Videos should be displayed correctly when loading is true 1`] = `ReactWrapper {}`;
